arcaflow_plugin_fio/fio_schema.py

```diff
@@ -58,108 +58,357 @@ def is_sync(self) -> bool:
+            "location than './'. You can specify a number of directories by separating "
+            "the names with a ':' character. These directories will be assigned "
+            "equally distributed to job clones created by numjobs as long as they are "
+            "using generated filenames. If specific filename(s) are set fio will use "
+            "the first listed directory, and thereby matching the filename semantic "
+            "(which generates a file for each clone if not specified, but lets all "
+            "clones use the same file if set)."
+        ),
+    ] = None
+    filename: typing.Annotated[
+        typing.Optional[str],
+        schema.name("Job Directory"),
+        schema.description(
+            "Fio normally makes up a filename based on the job name, thread number, "
+            "and file number. If you want to share files between threads in a job or "
+            "several jobs with fixed file paths, specify a filename for each of them "
+            "to override the default. If the ioengine is file based, you can specify a "
+            "number of files by separating the names with a ':' colon. So if you "
+            "wanted a job to open '/dev/sda' and '/dev/sdb' as the two working files, "
+            "you would use 'filename=/dev/sda:/dev/sdb'. This also means that whenever "
+            "this option is specified, nrfiles is ignored. The size of regular files "
+            "specified by this option will be size divided by number of files unless "
+            "an explicit size is specified by filesize."
+        ),
+    ] = None
+    nrfiles: typing.Annotated[
+        typing.Optional[int],
+        validation.min(1),
+        schema.name("Number of Files"),
+        schema.description(
+            "Number of files to use for this job. The size of files will be size "
+            "divided by this unless explicit size is specified by filesize. Files are "
+            "created for each thread separately, and each file will have a file number "
+            "within its name by default, as explained in filename section."
+        ),
+    ] = None
+    openfiles: typing.Annotated[
+        typing.Optional[int],
+        validation.min(1),
+        schema.name("Concurrent Open Files"),
+        schema.description(
+            "Number of files to keep open at the same time. Defaults to the same as "
+            "nrfiles, but can be set smaller to limit the number simultaneous opens."
+        ),
+    ] = None
+    create_on_open: typing.Annotated[
+        typing.Optional[int],
+        validation.min(0),
+        validation.max(1),
+        schema.name("Create Files on Open"),
+        schema.description(
+            "Don't pre-create files but allow the job's open() to create a file when "
+            "it's time to do I/O."
+        ),
+    ] = None
+    pre_read: typing.Annotated[
+        typing.Optional[int],
+        validation.min(0),
+        validation.max(1),
```

Comment:
As noted in my response to @jaredoconnell, I've been debating the value of changes like this. I agree this is better implemented as a `bool`, but that does cause more deviation of our schema from the fio input that is well-known by its users.

Reply:
Since the schema accepts integers as input to the `bool` type, it looks like I just need to add some logic to ensure that the value of the `bool` gets passed to the final config file as an `int` and then the problem is solved.